src/parameterizations/vertical/MOM_set_viscosity.F90

```diff
@@ -820,41 +821,90 @@ subroutine set_viscous_BBL(u, v, h, tv, visc, G, GV, CS)
 
 end subroutine set_viscous_BBL
 
-function set_v_at_u(v, h, G, i, j, k)
+function set_v_at_u(v, h, G, i, j, k, OBC)
   type(ocean_grid_type),                     intent(in) :: G
   real, dimension(SZI_(G),SZJB_(G),SZK_(G)), intent(in) :: v
   real, dimension(SZI_(G),SZJ_(G),SZK_(G)),  intent(in) :: h
   integer,                                   intent(in) :: i, j, k
+  type(ocean_OBC_type), pointer, optional               :: OBC
   real                                                  :: set_v_at_u
+  integer                                            :: n
   ! This subroutine finds a thickness-weighted value of v at the u-points.
   real :: hwt(4)           ! Masked weights used to average v onto u, in H.
   real :: hwt_tot          ! The sum of the masked thicknesses, in H.
+  type(OBC_segment_type), pointer :: segment
 
   hwt(1) = (h(i,j-1,k) + h(i,j,k)) * G%mask2dCv(i,J-1)
   hwt(2) = (h(i+1,j-1,k) + h(i+1,j,k)) * G%mask2dCv(i+1,J-1)
   hwt(3) = (h(i,j,k) + h(i,j+1,k)) * G%mask2dCv(i,J)
   hwt(4) = (h(i+1,j,k) + h(i+1,j+1,k)) * G%mask2dCv(i+1,J)
+
+
+  if (present(OBC)) then
```

Comment:
I see what @Hallberg-NOAA was talking about (looping over OB segments for each point). This is a case where checking the segment number via `OBC%OBC_segment_u(i,j)` is still much more efficient. Generally looping over the segments and then doing the i- or j- is fastest but in this case, since we are deep inside an i-j loop already I think there is little choice?